src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -736,9 +736,12 @@ private void recover()
           }
         }
       );
-      // batch call Snowpipe to ingest file
-      ingestionService.ingestFiles(filesForIngestion);
-      
+
+      if (!filesForIngestion.isEmpty())
```

Comment:
Can you put this check inside ingestionService.ingestFiles?

Reply:
Thanks for the feedback. However, I'm not sure I can agree with your suggestion as I feel this would break a behavioural contract with the *Service API.

_If_ the `SnowflakeIngestionServiceV1` should handle faulty input, it would imho make sense to raise expections on invalid input instead (empty-Set is invalid input here), and subsequently, any invoking client would need to make sure to call the service in a valid way (either by checking themself, as is done here, or in some other form, such as wrapping the invocation with `try..catch..`.

It should be the responsibility of orchestrating code to either invoke or not invoke the service, instead of silent-dropping any such request.

What do you think?

Reply:
That makes sense to me. Thanks a lot for your PR. However, we are having a major restructure going on right now, which completely removed this recoverFileStatus function. I'll notify you once that is done.